Src/Metrics/PerfCounters/PerformanceCounterGauge.cs

```diff
@@ -17,10 +17,13 @@ public PerformanceCounterGauge(string category, string counter, string instance)
         {
             try
             {
-                this.performanceCounter = instance == null ?
-                    new PerformanceCounter(category, counter, true) :
-                    new PerformanceCounter(category, counter, instance, true);
-                Metric.Internal.Counter("Performance Counters", Unit.Custom("Perf Counters")).Increment();
+                if (OperatingSystem.IsWindows())
```

Comment:
This should be solved more elegantly somehow - e.g. not letting you to configure performance counter metrics in the first place if not on windows. Or adding Linux support. Not necessarily in this PR, for the future.

Reply:
the problem here was that the code wouldn't compile because it signaled that the code can be reached from all platforms. this was a quick nasty fix. the better fix would be to add linux support, will do in another pr